app/controllers/dor_controller.rb

```diff
@@ -20,45 +15,33 @@ def reindex_from_cocina
     cocina_with_metadata = build_model_and_metadata(cocina_hash: cocina_hash,
                                                     created_at: params[:created_at].presence,
                                                     updated_at: params[:updated_at].presence)
-    druid = cocina_with_metadata.externalIdentifier
-    reindex_object(Success(cocina_with_metadata))
-    render status: :ok, plain: "Successfully updated index for #{druid}"
-  rescue CocinaModelBuildError => e
+    Indexer.reindex(solr: solr, cocina_with_metadata: cocina_with_metadata, commit_within: commit_within_param)
+    render status: :ok, plain: "Successfully updated index for #{cocina_with_metadata.externalIdentifier}"
+  rescue Cocina::Models::Error => e
```

Comment:
Just catching the actual errors instead of catching StandardError (below) and raising a custom error make this cleaner.